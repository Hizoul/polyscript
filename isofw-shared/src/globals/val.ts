const val = {
  isDebug: true,
  version: "0.1.0",
  i18nNameSpace: "langSpace",
  channel: {
    realtime: "realtime"
  },
  service: {
    user: "/users",
    project: "projects",
    camera: "cameras",
    preset: "presets",
    presetAssistant: "presetAssistant"
  },
  maximumPresetAmount: 100
}

export default val
