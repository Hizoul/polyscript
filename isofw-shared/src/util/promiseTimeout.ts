const promiseTimeout = (waitTime: number) => {
  return new Promise((resolve) => {setTimeout(resolve, waitTime)})
}
export default promiseTimeout
