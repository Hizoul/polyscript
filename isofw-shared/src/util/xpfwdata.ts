import {
  addId, AuthForm, BackendClient, dataOptions, DbStore, displayModeChanger,
  getListFormFromRelationshipField, IBackendClient, ICreateHookProps, IEditHookProps,
  IGetHookProps, IListHookProps, IRelationshipHookProps, IRemoveHookProps, IUiClient,
  ListStore, MailField, OwnerField, PwField, REMOVE_ADDON_KEY, removeId, searchRelated,
  useAuth, useCreate, useCreateWithProps, useEdit, useEditWithProps, useGet, useGetWithProps, toJS,
  useList, useListWithProps, useRelationship, useRelationshipWithProps, useRemove, useRemoveWithProps, UserStore
} from "@xpfw/data"

export {
  addId, AuthForm, BackendClient, dataOptions, DbStore, displayModeChanger,
  getListFormFromRelationshipField, IBackendClient, ICreateHookProps, IEditHookProps,
  IGetHookProps, IListHookProps, IRelationshipHookProps, IRemoveHookProps, IUiClient,
  ListStore, MailField, OwnerField, PwField, REMOVE_ADDON_KEY, removeId, searchRelated,
  useAuth, useCreate, useCreateWithProps, useEdit, useEditWithProps, useGet, useGetWithProps, toJS,
  useList, useListWithProps, useRelationship, useRelationshipWithProps, useRemove, useRemoveWithProps, UserStore
}
