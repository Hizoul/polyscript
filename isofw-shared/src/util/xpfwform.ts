import {
  ComponentRegistry, ExtendedJSONSchema, FormStore, FormStoreClass, getMapTo, getMapToFromProps, IFieldProps,
  iterateSubFields, JSONSchemaDefinition, jsonValidator, memo, prefixMaker, prependPrefix, SharedField,
  useArray, useField, useFieldWithValidation, useObject
} from "@xpfw/form"

export {
  ComponentRegistry, ExtendedJSONSchema, FormStore, FormStoreClass, getMapTo, getMapToFromProps, IFieldProps,
  iterateSubFields, JSONSchemaDefinition, jsonValidator, memo, prefixMaker, prependPrefix, SharedField,
  useArray, useField, useFieldWithValidation, useObject
}
