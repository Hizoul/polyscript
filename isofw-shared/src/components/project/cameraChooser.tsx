import { DbStore } from "isofw-shared/src/util/xpfwdata"
import { ExtendedJSONSchema, FormStore, getMapTo, memo, prependPrefix } from "isofw-shared/src/util/xpfwform"
import { PresetAssistantForm, PresetCameraField, PresetProjectField } from "isofw-shared/src/xpfwDefs/preset"
import { ProjectCameras, ShotPreset } from "isofw-shared/src/xpfwDefs/project"
import { flow } from "mobx"

const popupVisibilityKey = "cameraChoice"
const freePresetKey = "freePresetGetter"

const togglePop = (schema: ExtendedJSONSchema, mapTo?: string, prefix?: string, newValue?: any) => {
  if (mapTo == null) { mapTo = getMapTo(schema, mapTo)}
  return () => {
    FormStore.setValue(mapTo, newValue, prependPrefix(prefix, popupVisibilityKey))
  }
}
const untypedDbStore: any = DbStore

const setValueWithPreset = (schema: ExtendedJSONSchema, mapTo?: any, prefix?: string) => {
  if (mapTo == null) { mapTo = getMapTo(schema, mapTo)}
  return flow(function *(newValue?: any) {
    const creationPrefix = prependPrefix(`${prependPrefix(PresetAssistantForm.title, freePresetKey)}`, prefix)
    FormStore.setValue(getMapTo(schema, mapTo), newValue, prefix)
    FormStore.setValue(PresetAssistantForm.title, {}, prependPrefix(freePresetKey, prefix))
    FormStore.setValue(PresetCameraField.title, newValue, creationPrefix)
    FormStore.setValue(PresetProjectField.title, untypedDbStore.currentlyEditing, creationPrefix)
    const freeId: any = yield DbStore.create(PresetAssistantForm, undefined, prependPrefix(freePresetKey, prefix))
    mapTo = mapTo.substring(0, mapTo.indexOf("]") + 1)
    FormStore.setValue(`${prependPrefix(mapTo, prefix)}.${ShotPreset.title}`, freeId)
    return freeId
  })
}

const emptyArray = (size: number) => {
  const arr: any[] = []
  for (let index = 0; index < size; index++) {
    arr.push(undefined)
  }
  return arr
}

const useCameraChooser = (schema: ExtendedJSONSchema, mapTo?: string, prefix?: string) => {
  if (mapTo == null) { mapTo = getMapTo(schema, mapTo) }
  let cameras = FormStore.getValue(ProjectCameras.title, prefix)
  cameras = cameras ? cameras : []
  const arr = mapTo.substring(0, mapTo.indexOf("["))
  const pop = FormStore.getValue(arr, prependPrefix(prefix, popupVisibilityKey))
  if (pop == null || pop.length < 600) {
    FormStore.setValue(arr, emptyArray(602), prependPrefix(prefix, popupVisibilityKey))
  }
  return {
    cameras,
    value: FormStore.getValue(getMapTo(schema, mapTo), prefix),
    showPopUp: FormStore.getValue(getMapTo(schema, mapTo), prependPrefix(prefix, popupVisibilityKey), false) === true,
    hidePop: memo(() => togglePop(schema, mapTo, prefix, false), ["hidePop", mapTo, prefix, JSON.stringify(schema)]),
    showPop: memo(() => togglePop(schema, mapTo, prefix, true), ["showPop", mapTo, prefix, JSON.stringify(schema)]),
    setValueWithPreset: memo(() => setValueWithPreset(schema, mapTo, prefix), ["setValueWithPreset", JSON.stringify(schema), mapTo, prefix])
  }
}

export default useCameraChooser
export { togglePop, setValueWithPreset }
