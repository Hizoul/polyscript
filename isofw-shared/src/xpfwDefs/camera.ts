import { MailField, OwnerField, PwField } from "@xpfw/ui-shared"
import ValidationRegistry, { FieldType, IField, IForm, Permission } from "@xpfw/validate"
import collections from "isofw-shared/src/xpfwDefs/collections"
import { ProjectName } from "./project";

const CameraIp: IField = {
  type: FieldType.Text,
  mapTo: "ip"
}

const CameraForm: IForm = {
  model: "cameraModel",
  collection: "cameras",
  sections: [{fields: [
    ProjectName, CameraIp
  ]}],
  permissions: {
    required: {
      create: Permission.User,
      find: Permission.User,
      get: Permission.User,
      update: Permission.User,
      remove: Permission.User
    }
  },
  options: {
    addCreatedAt: true,
    idPath: "_id"
  }
}

ValidationRegistry.registerForm(CameraForm)

export {
  CameraForm, CameraIp
}