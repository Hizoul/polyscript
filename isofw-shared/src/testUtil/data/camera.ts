import isServerParams from "isofw-shared/src/globals/isServerParams"
import val from "isofw-shared/src/globals/val"
import { CameraIp } from "isofw-shared/src/xpfwDefs/camera"
import { ProjectName } from "isofw-shared/src/xpfwDefs/project"

const testCameras = [
  {_id: "c1bbbbbbbbbbbbbbbbbbbbbb", [String(ProjectName.title)]: "Camera#1", [String(CameraIp.title)]: "localhost:4202/c1"},
  {_id: "c2bbbbbbbbbbbbbbbbbbbbbb", [String(ProjectName.title)]: "Camera#2", [String(CameraIp.title)]: "localhost:4202/c2"},
  {_id: "c3bbbbbbbbbbbbbbbbbbbbbb", [String(ProjectName.title)]: "Camera#3", [String(CameraIp.title)]: "localhost:4202/c3"}
]
8
const createTestCameras = async (app: any) => {
  const newCameras = []
  for (const newUser of testCameras) {
    newCameras.push(await app.service(val.service.camera).create(newUser, isServerParams))
  }
  return newCameras
}

export default createTestCameras
export {
  testCameras
}
