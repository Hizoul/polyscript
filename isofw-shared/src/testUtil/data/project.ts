import { MailField, PwField } from "@xpfw/ui-shared"
import val from "isofw-shared/src/globals/val"
import isServerParams from "isofw-shared/src/globals/isServerParams"
import { ProjectForm, ProjectName, ProjectShot } from "isofw-shared/src/xpfwDefs/project";

const testProjects = [
  {_id: "b19999999999999999999999", [ProjectName.mapTo]: "pn1", [ProjectShot.mapTo]: 1},
  {_id: "b29999999999999999999999", [ProjectName.mapTo]: "pn2", [ProjectShot.mapTo]: 8},
  {_id: "b39999999999999999999999", [ProjectName.mapTo]: "pn3", [ProjectShot.mapTo]: 6}
]
8
const createTestProjects = async (app: any) => {
  const newProjects = []
  for (const newUser of testProjects) {
    newProjects.push(await app.service(val.service.project).create(newUser, isServerParams))    
  }
  return newProjects
}

export default createTestProjects
export {
  testProjects
}