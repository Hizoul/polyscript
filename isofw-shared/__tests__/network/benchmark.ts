import FeathersClient from "@xpfw/data-feathers"
import { makeSubFields } from "@xpfw/form-tests"
import TCPClient from "isofw-node/src/network/tcpClient"
import UDPClient from "isofw-node/src/network/udpClient"
import getNetworkTestApp from "isofw-node/src/testUtil/getNetworkTestApp"
import { setValueWithPreset } from "isofw-shared/src/components/project/cameraChooser"
import val from "isofw-shared/src/globals/val"
import makeBenchmarkClient from "isofw-shared/src/network/clientBenchmarker"
import createTestCameras from "isofw-shared/src/testUtil/data/camera"
import createTestProjects from "isofw-shared/src/testUtil/data/project"
import createTestUsers from "isofw-shared/src/testUtil/data/users"
import logIntoUser from "isofw-shared/src/testUtil/login"
import promiseTimeout from "isofw-shared/src/util/promiseTimeout"
import { BackendClient, DbStore, ListStore, toJS, UserStore } from "isofw-shared/src/util/xpfwdata"
import { FormStore } from "isofw-shared/src/util/xpfwform"
import { CameraIp } from "isofw-shared/src/xpfwDefs/camera"
import {
  PresetAssistantForm, PresetCameraField, PresetForm, PresetProjectField
} from "isofw-shared/src/xpfwDefs/preset"
import { ProjectName } from "isofw-shared/src/xpfwDefs/project"

const untypedDbStore: any = DbStore
describe("network benchmark test", () => {
  it("should collect numeric data", async () => {
    const configs = [
      {network: val.network.tcp, client: TCPClient},
      {network: val.network.udp, client: UDPClient},
      {network: val.network.websocket, client: FeathersClient}
    ]
    for (const config of configs) {
      const clientToUse = makeBenchmarkClient(config.client)
      BackendClient.client = clientToUse
      const tcpServer = await getNetworkTestApp(config.network, clientToUse)
      await createTestUsers(tcpServer.app)
      expect(toJS(UserStore)).toMatchSnapshot("Before Login")
      await logIntoUser()
      expect(toJS(UserStore)).toMatchSnapshot("After Login")
      expect(toJS(DbStore)).toMatchSnapshot("After Log in but before real-time")
      await tcpServer.app.service(val.service.camera).create({
        _id: "c9bbbbbbbbbbbbbbbbbbbbbb",
        [String (ProjectName.title)]: " real-time created camera ",
        [String (CameraIp.title)]: "What an IP"
      })
      const cameraResult = await createTestCameras(tcpServer.app)
      const projectResults = await createTestProjects(tcpServer.app, true)
      ListStore.pageSize = 400
      await ListStore.getList(PresetForm, undefined, undefined, true)
      await tcpServer.cleanUp()
      await promiseTimeout(1000)
      console.log("THE MEASUREMENTS ARE", clientToUse.measurements)
      expect(clientToUse.measurements.length).not.toBe(0)
    }

  }, 160000)
})
