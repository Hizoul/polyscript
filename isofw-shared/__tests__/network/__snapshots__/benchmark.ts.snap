// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`network benchmark test should collect numeric data: After Log in but before real-time 1`] = `
Object {
  "createState": Object {},
  "currentlyEditing": "",
  "fetching": Object {},
  "getFromServer": [Function],
  "getState": Object {
    "users": Object {
      "-1": undefined,
    },
  },
  "lastFetch": Object {},
  "removeState": Object {},
  "updateState": Object {},
}
`;

exports[`network benchmark test should collect numeric data: After Log in but before real-time 2`] = `
Object {
  "createState": Object {},
  "currentlyEditing": "",
  "fetching": Object {},
  "getFromServer": [Function],
  "getState": Object {
    "users": Object {
      "-1": undefined,
    },
  },
  "lastFetch": Object {},
  "removeState": Object {},
  "updateState": Object {},
}
`;

exports[`network benchmark test should collect numeric data: After Login 1`] = `
Object {
  "authErr": null,
  "connected": true,
  "loading": false,
  "loggedIn": true,
  "user": undefined,
}
`;

exports[`network benchmark test should collect numeric data: After Login 2`] = `
Object {
  "authErr": Object {},
  "connected": true,
  "loading": false,
  "loggedIn": false,
  "user": null,
}
`;

exports[`network benchmark test should collect numeric data: Before Login 1`] = `
Object {
  "authErr": null,
  "connected": true,
  "loading": false,
  "loggedIn": false,
  "user": null,
}
`;

exports[`network benchmark test should collect numeric data: Before Login 2`] = `
Object {
  "authErr": null,
  "connected": true,
  "loading": false,
  "loggedIn": true,
  "user": undefined,
}
`;
